### DISCUSSION TITLE
How to prune the inverted index

### DISCUSSION THREAD

**Discussion by musalbas** [celestiaorg/celestia-node#2647] [General]
See https://github.com/celestiaorg/celestia-node/issues/2615

**Comment by musalbas** [comment]
I think it's definitely possible to implement inverted index pruning purely in inverted_index.go by modifying AddMultihashesForShard and GetShardsForMultihash. Here's how:
* Let's say storage window is 30 days. We keep a badger index for the last 30 days of data; and one for the 30 days prior (60 days total).
* By construction, AddMultihashesForShard only ever needs to add indexes to blocks in the most recent badger index, thus no height parameter is needed in the call stack.
* GetShardsForMultihash needs to try both indexes anyway -- doing up to two reads.
* There should be some background routine that deletes the old index automatically once we've accumulated >30 days' worth of blocks.

**Reply by distractedm1nd** [reply]
@Wondertan is there any possibility of a CID being found in two distinct blocks? I don't think so but I may be forgetting something, like padding

**Reply by Wondertan** [reply]
In the case of padding, there is the same share in almost every block which will have the same indexed CID.
